### ISSUE TITLE
class methods and properties which have the same name as imported modules cause type check errors

### ISSUE THREAD

**Issue by plafosse** [microsoft/pyright#1992]
**Describe the bug**
This might need to be filed under the "can't fix" category. I'm unclear if there is a known work around... Given the following code
`foo.py`
```python
bar = int
```

and `test.py`

```python
from . import foo

class Bas():
    def __init__(self, a:'foo.bar'):
        self.a = a

    def foo(self):
        return None
```

pyright produces the following error

```
Expected class type but received "(self: Bas) -> None"
```

Additionally if foo is defined as a property the error is slightly different.

```
Expected class type but received "property"
```

It's obvious that the type checker is confusing the import and the class method because they have the same name.  

**Expected behavior**
Properly disambiguate between modules and methods.

**VS Code extension or command-line**
This is being run as part of the VS Code extension

**Comment by erictraut** [comment]
Which version of pyright are you running? I fixed a bug related to this one or two versions ago, and I'm not able to repro it with the latest version. Please verify that you're running 1.1.149.

**Comment by plafosse** [comment]
Its the one installed in with VSCode Pylance which seems to still be on 1.1.148

**Comment by plafosse** [comment]
I attempted to install Pyright VSCode plugin directly but it complained that it would conflict with Pylance's version and auto disabled it. Is there a way to run 1.1.149 and Pylance at the same time?

**Comment by erictraut** [comment]
The screenshot indicates that the problem is that the import could not be resolved. I'm guessing that `foo.py` is not in the same directory as `test.py`. If you place the two in the same directory, I think you'll find that it works.

The other bug was fixed since last week's release of pylance, and the fix will be included in this week's release.

**Comment by plafosse** [comment]
Sorry my screenshot wasn't clear. The files are in the same directory. I also added an `__init__.py` just to see if that made a difference. Additionally it's clear that it is resolving the 'foo.bar' in tje context of the member variable assignment in this image. 
![image](https://user-images.githubusercontent.com/10556585/122263836-dfd26c80-cea4-11eb-860c-a71332c5d1a2.png)

Yet its still confused about the type in the parameter.
![image](https://user-images.githubusercontent.com/10556585/122263982-085a6680-cea5-11eb-8dc4-a642ea3df70f.png)

**Comment by erictraut** [comment]
Ah, I'm able to repro now. I had removed the quotes around `'foo.bar'` when typing in the code from your screen shot. 

This is expected behavior. You have enclosed the type annotation in quotes, which means that it's a forward reference. It refers in this case to the `foo` symbol declared in the `Bas` class, not the `foo` symbol imported in the outer scope. You have a couple of options:

1. Remove the quotes from the annotation.
2. Rename `foo` to avoid the namespace collision:

```python
from . import foo as _foo

class Bas:
    def __init__(self, a: '_foo.bar'):
        self.a = a

    def foo(self):
        return None
```

**Comment by plafosse** [comment]
Ok I was aware of those workarounds, I just didn't like either of them as it's a rather large project :P  The quotes are necessary in my case due to circular references. I guess I'm just stuck with aliasing the import name.  Thank you so much for your help and responsiveness, and making such a great tool!
